ejb3/src/main/java/org/jboss/as/ejb3/component/stateful/StatefulSessionComponentInstance.java

```diff
@@ -72,6 +75,9 @@
      */
     private boolean synchronizationRegistered = false;
 
+    private final AtomicReference<Boolean> delayedAfterCompletion = new AtomicReference<>(false);
```

Comment:
I used AtomicReference<Boolean>, opposed to AtomicBoolean, as we required a tri-state, the idea being that null represents the third state. However, I agree that it would be better to have a single entity that manages both states.